src/status_im/notifications/core.cljs

```diff
@@ -216,17 +217,21 @@
       (log/debug "handle-on-message" "app-state:" app-state
                  "view-id:" view-id "current-chat-id:" current-chat-id
                  "from:" from "force:" force)
-      (when (or force
-                (and
-                 (not= app-state "active")
-                 (show-notification? cofx rehydrated-payload)))
-        {:db
-         (assoc-in db [:push-notifications/stored (:to rehydrated-payload)]
-                   (js/JSON.stringify (clj->js rehydrated-payload)))
-         :notifications/display-notification
-         {:title           (i18n/label :notifications-new-message-title)
-          :body            (i18n/label :notifications-new-message-body)
-          :decoded-payload rehydrated-payload}})))
+      (merge
+       (when (and (= (.-length from) pn-pubkey-length)
+                  (show-notification? cofx rehydrated-payload))
+         {:dispatch [:mailserver/fetch-history from (- (quot now 1000) pull-recent-messages-window)]})
```

Comment:
We should be a bit careful here, you want to make sure that we are not overriding an earlier value of from.

Say we just logged in, and `from` is -1 day, and we haven't yet fetched stuff from the mailserver, we want to make sure that `from` is `(min your-from current-from)`, do you see what I mean? (sorry if it's a bit confusing)

Reply:
If I understand correctly, this is what you're suggesting:

``` clojure
(let [fetch-pn-message-window (- (quot now 1000) pull-recent-messages-window)
      last-updated            (get-in db [:account/account :last-updated])]
  {:dispatch [:mailserver/fetch-history from (min last-updated fetch-pn-message-window)]}))
```

Reply:
OK, fixed.